espnet2/train/preprocessor.py

```diff
@@ -944,3 +946,236 @@ def _speech_process(
+        align: list = [
+            "singing",
+            "label_lab",
+            "midi_lab",
+            "tempo_lab",
+            "beat_lab",
+        ],  # TODO(Tao): add to args
+    ):
+        super().__init__(train)
+        self.train = train
+        self.singing_name = singing_name
+        self.text_name = text_name
+        self.label_name = label_name
+        self.midi_name = midi_name
+        self.fs = fs
+        self.singing_volume_normalize = singing_volume_normalize
+        self.align = align
+        if token_type is not None:
+            if token_list is None:
+                raise ValueError("token_list is required if token_type is not None")
+            self.text_cleaner = TextCleaner(text_cleaner)
+
+            self.tokenizer = build_tokenizer(
+                token_type=token_type,
+                bpemodel=bpemodel,
+                delimiter=delimiter,
+                space_symbol=space_symbol,
+                non_linguistic_symbols=non_linguistic_symbols,
+                g2p_type=g2p_type,
+            )
+            self.token_id_converter = TokenIDConverter(
+                token_list=token_list,
+                unk_symbol=unk_symbol,
+            )
+        else:
+            self.text_cleaner = None
+            self.tokenizer = None
+            self.token_id_converter = None
+
+    def __call__(
+        self,
+        uid: str,
+        data: Dict[str, Union[str, np.ndarray, tuple]],
+    ) -> Dict[str, np.ndarray]:
+        assert check_argument_types()
+
+        if self.singing_name in data:
+            if self.singing_volume_normalize is not None:
+                singing = data[self.singing_name]
+                ma = np.max(np.abs(singing))
+                data[self.singing_name] = singing * self.singing_volume_normalize / ma
+
+        if (
+            self.midi_name in data
+            and self.label_name in data
+            and self.tokenizer is not None
+        ):
+            # Load label info
+            lab_timeseq, text = data[self.label_name]
+            lab_len = len(text)
+            text = " ".join(text)
+            text = self.text_cleaner(text)
+            tokens = self.tokenizer.text2tokens(text)
+            text_ints = self.token_id_converter.tokens2ids(tokens)
+            data.pop(self.label_name)
+
+            # Load xml info
+            # TODO(Yuning): It can only work on Japanese dataset now.
+            # Syllable2phoneme mismatch and more languags settings need to be settled.
+            syllables, notemidis, notetimeseq, tempo = data[self.midi_name]
+            midis = []
+            xml_timeseq = []
+            phn_cnt = 0
+            vowel = ["a", "i", "u", "e", "o"]
+            for i in range(len(syllables)):
+                # NOTE: Some phonemes are tagged differently
+                if syllables[i] == "へ":
+                    phn = ["h", "e"]
+                else:
+                    phn = pyopenjtalk.g2p(syllables[i]).split(" ")
```

Comment:
can we instead use `self.tokenizer` to do this part? The current procedure breaks the CI at https://github.com/espnet/espnet/actions/runs/3141052108/jobs/5103071721 . @A-Quarter-Mile